controllers/pulp/controller.go

```diff
@@ -58,7 +58,7 @@ type PulpReconciler struct {
 //+kubebuilder:rbac:groups=config.openshift.io,resources=ingresses,verbs=get;list;watch
 //+kubebuilder:rbac:groups=route.openshift.io,resources=routes;routes/custom-host,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;
-//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles;rolebindings;serviceaccounts,verbs=create;update;patch;delete;watch;get;list;
+//+kubebuilder:rbac:groups=core;rbac.authorization.k8s.io,resources=roles;rolebindings;serviceaccounts,verbs=create;update;patch;delete;watch;get;list;
```

Comment:
not sure if this is the proper fix for the serviceaccounts issue

Reply:
hum... sounds like it is:
```
$ oc explain serviceaccount
KIND:     ServiceAccount
VERSION:  v1
...

$ oc explain roles
KIND:     Role
VERSION:  rbac.authorization.k8s.io/v1
...
```

Reply:
actually it was not enough:
```
2022-09-16T11:15:28Z	[35mDEBUG[0m	pulp/controller.go:72	Running on OpenShift cluster
2022-09-16T11:15:28Z	[34mINFO[0m	pulp/rbac.go:63	Creating a new pulp Role	{"Namespace": "openshift-operators", "Name": "example"}
2022-09-16T11:15:28Z	[31mERROR[0m	pulp/rbac.go:67	Failed to create new pulp ServiceAccount	{"Namespace": "openshift-operators", "Name": "example", "error": "roles.rbac.authorization.k8s.io \"example\" is forbidden: user \"system:serviceaccount:openshift-operators:pulp-operator-controller-manager\" (groups=[\"system:serviceaccounts\" \"system:serviceaccounts:openshift-operators\" \"system:authenticated\"]) is attempting to grant RBAC permissions not currently held:\n{APIGroups:[\"\"], Resources:[\"pods/log\"], Verbs:[\"get\"]}"}
github.com/pulp/pulp-operator/controllers/pulp.(*PulpReconciler).CreateRole
	/workspace/controllers/pulp/rbac.go:67
github.com/pulp/pulp-operator/controllers/pulp.(*PulpReconciler).CreateServiceAccount
	/workspace/controllers/pulp/rbac.go:54
github.com/pulp/pulp-operator/controllers/pulp.(*PulpReconciler).Reconcile
	/workspace/controllers/pulp/controller.go:144
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234
2022-09-16T11:15:28Z	[31mERROR[0m	controller/controller.go:273	Reconciler error	{"controller": "pulp", "controllerGroup": "repo-manager.pulpproject.org", "controllerKind": "Pulp", "pulp": {"name":"example","namespace":"openshift-operators"}, "namespace": "openshift-operators", "name": "example", "reconcileID": "5d272658-2982-4f0d-a5c9-f30da0dba271", "error": "roles.rbac.authorization.k8s.io \"example\" is forbidden: user \"system:serviceaccount:openshift-operators:pulp-operator-controller-manager\" (groups=[\"system:serviceaccounts\" \"system:serviceaccounts:openshift-operators\" \"system:authenticated\"]) is attempting to grant RBAC permissions not currently held:\n{APIGroups:[\"\"], Resources:[\"pods/log\"], Verbs:[\"get\"]}"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234
2022-09-16T11:15:28Z	[35mDEBUG[0m	events	record/event.go:314	Warning	{"object": {"kind":"Pulp","namespace":"openshift-operators","name":"example","uid":"5de678c0-c959-440e-bd81-a0cb0eb835d4","apiVersion":"repo-manager.pulpproject.org/v1alpha1","resourceVersion":"35378"}, "reason": "Failed", "message": "Failed to create new pulp Role"}
```

Reply:
I've added pods/log permission, and it finally works!